src/data/fin.lean

```diff
@@ -121,6 +123,12 @@ rfl
 @[simp] lemma cast_succ_inj {a b : fin n} : a.cast_succ = b.cast_succ ↔ a = b :=
 by simp [eq_iff_veq]
 
+def with_max (n m : ℕ) (h : m > 0) : fin m :=
+(fin.of_nat (min n (nat.pred m))).cast $ nat.succ_pred_eq_of_pos h
+
+@[simp] lemma with_max_val (n m : ℕ) (h : m > 0) : (with_max n m h).val = min n (m - 1) :=
```

Comment:
Would it not be nicer to consider values in `fin $ n+1`? This way, you don’t need a proof term and your law becomes `(with_max m n).val = min m n

Reply:
I find the name `with_max` a bit confusing. Maybe `with_ceiling` or `with_ceil` would be better.

Reply:
how about `clamp`?

Reply:
Are you by any chance a fan of Futurama?